ext/mysqlnd/mysqlnd_loaddata.c

```diff
@@ -65,11 +65,10 @@ static
 int mysqlnd_local_infile_read(void * ptr, zend_uchar * buf, unsigned int buf_len)
 {
 	MYSQLND_INFILE_INFO	*info = (MYSQLND_INFILE_INFO *)ptr;
-	int count;
 
 	DBG_ENTER("mysqlnd_local_infile_read");
 
-	count = (int) php_stream_read(info->fd, (char *) buf, buf_len);
+	ssize_t count = php_stream_read(info->fd, (char *) buf, buf_len);
```

Comment:
The return type of `int mysqlnd_local_infile_read()` should be changed as well, otherwise the `count < 0` check bellow may be false, but the return value may still be `< 0` due to conversion to `int`.

Reply:
Where does the conversion to int take place?

Reply:
It happens when returning `count`, as the return type is `int`

Reply:
It's an unrelated change so it should be probably removed from this PR and different PR created for this.